src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs

```diff
@@ -112,6 +126,7 @@ public async Task UpdateInstanceBlobsAsync([ActivityTrigger] UpdateInstanceBlobA
         }
 
         logger.LogInformation("Completed updating all instance blobs");
+        return new ReadOnlyCollection<WatermarkedFileProperties>(propertiesByWatermark);
```

Comment:
`List<T>` implements `IReadOnlyList<T>` so you don't need this class. Just return `propertiesByWatermark`